src/hotspot/share/gc/shenandoah/shenandoahGeneration.cpp

```diff
@@ -228,8 +229,19 @@ void ShenandoahGeneration::merge_write_table() {
 }
 
 void ShenandoahGeneration::prepare_gc() {
+  ShenandoahHeap* const heap = ShenandoahHeap::heap();
+  if (need_bitmap_reset()) {
+    if (heap->mode()->is_generational() && is_global() && !heap->young_generation()->need_bitmap_reset()) {
+      //Only need to reset bitmap for old generation.
```

Comment:
I'd like to see an assert here that all the bitmaps for all the young regions are already clear.

Reply:
I'll add it, thanks.

Reply:
My understanding is: op_reset_after_collect is called at the end of a successful collection after the old region (A) is repurposed as a young region, if so it is not an issue, I'll double confirm.